packages/core/src/repository/data-source/network.data-source.ts

```diff
@@ -0,0 +1,102 @@
+import { lastValueFrom, Observable } from 'rxjs';
+import { DataSource } from './data-source';
+import { Query } from '../query/query';
+import { MethodNotImplementedError, QueryNotSupportedError } from '../errors';
+import { DataSourceMapper } from './data-source-mapper';
+import { ArrayMapper, JsonDeserializerMapper, VoidMapper } from '../mapper/mapper';
+import { Type } from '../../helpers';
+import {
+    DeleteNetworkQuery,
+    GetNetworkQuery,
+    NetworkQuery,
+    PostNetworkQuery,
+    PutNetworkQuery,
+} from '../query/network-query';
+import { ApiRequestService } from './api-request.service';
+import { HttpRequestBuilder } from '../../data';
+
+export class NetworkDataSource implements DataSource<unknown> {
+    constructor(private readonly requestService: ApiRequestService) {}
+
+    public get(query: Query): Promise<unknown> {
+        if (query instanceof GetNetworkQuery) {
+            return lastValueFrom(this.getRequestWithParameters<unknown>(query).get());
+        }
+
+        throw new QueryNotSupportedError();
+    }
+
+    public put(value: unknown | undefined, query: Query): Promise<unknown> {
+        let $request: Observable<unknown>;
+        if (query instanceof PutNetworkQuery) {
+            $request = this.getRequestWithParameters<void>(query).put();
+        } else if (query instanceof PostNetworkQuery) {
+            $request = this.getRequestWithParameters<void>(query).post();
+        } else {
+            throw new QueryNotSupportedError();
+        }
+
+        return lastValueFrom($request);
+    }
+
+    public delete(query: Query): Promise<void> {
+        if (query instanceof DeleteNetworkQuery) {
+            return lastValueFrom(this.getRequestWithParameters<void>(query).delete());
+        }
+
+        throw new QueryNotSupportedError();
+    }
+
+    private getRequestWithParameters<T extends unknown | void>(
+        query: NetworkQuery,
+        value?: unknown | undefined,
+    ): HttpRequestBuilder<T> {
+        const request = this.requestService
+            .builder<T>(query.endpoint)
+            .setQueryParameters(query.queryParameters)
+            .setUrlParameters(query.urlParameters);
```

Comment:
Open to discussion:

The idea of sharing this method on all types of requests comes from the http specifications, when we can have query parameters + url parameters + body in GET/POST/PUT/DELETE methods, so we have the most generic network datasource we can have.

Our agreement with the backend team is to never use URL parameters, and never use body in GET/DELETE methods. We could change this datasource to match that, but this will only work against our own backends.

Reply:
I would leave it as it is, as it's more flexible.